### PR TITLE
fix: remove Torchx compilation warnings

### PR DESCRIPTION
**Pull request by polvalente**
Two minor fixes which silence all warnings which were being output by Torchx on my machine during compilation

### CODE REVIEW

File: torchx/Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ LIBTORCH_API_INCLUDE_PATH = $(LIBTORCH_DIR)/include/torch/csrc/api/include
 LIBTORCH_LINK_PATH = $(LIBTORCH_DIR)/lib/
 CFLAGS = -fPIC -I$(ERTS_INCLUDE_DIR) -I$(LIBTORCH_INCLUDE_PATH) -I$(LIBTORCH_API_INCLUDE_PATH) \
 	 -O3 -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -shared \
-	 -std=c++14 -D_GLIBCXX_USE_CXX11_ABI=0
+	 -std=c++14 -D_GLIBCXX_USE_CXX11_ABI=0 -fopenmp
```
Comment (polvalente) [inline torchx/Makefile:6]: This is to handle a `pragma` directive

File: torchx/c_src/torchx.cpp [hunk 2]
```diff
@@ -58,7 +58,7 @@ inline std::string type2string(const torch::ScalarType type)
   }
 
 #define CATCH()                                              \
-  catch (c10::Error error)                                   \
+  catch (c10::Error &error)                                   \
```
Comment (polvalente) [inline torchx/c_src/torchx.cpp:61]: This one silences the vast majority of warnings, which all stemmed from a "catch by value" warning here.

**Review by josevalim — APPROVED** [review]

File: torchx/Makefile (hunk 1 above)
Comment (josevalim) [reply torchx/Makefile:6]: Actually, this option is not available on Mac. Should we move it to the Linux branch?

**Review by josevalim — CHANGES_REQUESTED** [review]
One minor comment.

File: torchx/Makefile (hunk 1 above)
Comment (polvalente) [reply torchx/Makefile:6]: Sure!

**Pull request merged**